compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -992,11 +980,44 @@ private STNode parseReturnTypeDescRhs(STNode metadata, STNode functionKeyword, S
         funcSignature = validateAndGetFuncParams((STFunctionSignatureNode) funcSignature);
 
         STNode funcDef =
-                createFuncDefOrMethodDecl(metadata, functionKeyword, isObjectMethod, name, funcSignature, qualifiers);
+                createFuncDefOrMethodDecl(metadata, functionKeyword, isObjectMember, name, funcSignature, qualifiers);
         endContext();
         return funcDef;
     }
 
+    private STNode parseVarDeclWithFunctionType(STNode typeDesc, boolean isObjectMember, STNode qualifiers,
+                                                STNode metadata) {
+        STNodeList qualifierList = (STNodeList) qualifiers;
+        if (isObjectMember) {
+            // public qualifier allowed in object field.
+            STNode readonlyQualifier = STNodeFactory.createEmptyNode();
+            STNode fieldName = parseVariableName();
+            if (qualifierList.isEmpty()) {
+                return parseObjectFieldRhs(metadata, STNodeFactory.createEmptyNode(), readonlyQualifier,
+                        typeDesc, fieldName);
+            } else {
+                return parseObjectFieldRhs(metadata, qualifiers.childInBucket(0), readonlyQualifier, typeDesc,
+                        fieldName);
+            }
+        }
+
+        STNode finalKeyword = STNodeFactory.createEmptyNode();
+        // Only the final keyword is allowed as a qualifier
+        for (int position = 0; position < qualifierList.size(); position++) {
```

Comment:
do we need the similar validation for object members too (i.e: if `isObjectMember` is true) ?

Reply:
+1 will update.